services/shuttle-crontab/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use axum::Router;
+use chrono::Utc;
+use cron::Schedule;
+use router::make_router;
+use serde::{Deserialize, Serialize};
+use shuttle_persist::PersistInstance;
+use shuttle_runtime::tracing::{debug, info};
+use tokio::{
+    sync::mpsc::{self, Receiver, Sender},
+    sync::oneshot,
+    time::sleep,
+};
+
+mod error;
+use error::CrontabServiceError;
+
+mod router;
+
+pub type ShuttleCrontab = Result<CrontabService, shuttle_runtime::Error>;
+
+type Responder<T> = oneshot::Sender<Result<T, CrontabServiceError>>;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RawJob {
+    schedule: String,
+    url: String,
+}
+
+#[derive(Debug)]
+enum Msg {
+    NewJob(RawJob, Responder<()>),
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Crontab {
+    jobs: Vec<RawJob>,
+}
+
+#[derive(Debug)]
+struct CronJob {
+    schedule: Schedule,
+    url: String,
+}
+
+impl CronJob {
+    fn from_raw(raw: &RawJob) -> Self {
```

Comment:
I would use the idiomatic implementation of `From<RawJob>`  for `CronJob`.

Reply:
Obviously. I think I had something else here and refactored and then didn't realise...